parlai/crowdsourcing/tasks/pairwise_per_turn_eval/analysis/compile_results.py

```diff
@@ -0,0 +1,553 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import os
+import re
+from collections import defaultdict
+from datetime import datetime
+from typing import Any, Dict
+
+import pandas as pd
+
+import parlai.utils.logging as logging  # TODO: check if this is how this import is usually done
+from parlai.crowdsourcing.utils.acceptability import AcceptabilityChecker
+from parlai.crowdsourcing.utils.analysis import AbstractResultsCompiler
+from parlai.crowdsourcing.tasks.pairwise_per_turn_eval.per_turn_eval_blueprint import (
+    BLUEPRINT_TYPE,
+)  # noqa: F401  # For registering the blueprint
+
+
+class PerTurnEvalResultsCompiler(AbstractResultsCompiler):
+    """
+    Compile and save results of human+model chats.
+
+    Results will be saved on the level of specific conversations, as well as aggregated
+    up the level of each worker as a whole.
+    """
+
+    # TODO: deduplicate setup_args from ModelChatResultsCompiler
+    @classmethod
+    def setup_args(cls):
+        parser = super().setup_args()
+        parser.add_argument(
+            '--worker-block-list',
+            type=str,
+            default='',
+            help='Comma-separated list of all workers to block',
+        )
+        return parser
+
+    def __init__(self, opt: Dict[str, Any]):
+        # TODO: deduplicate init from ModelChatResultsCompiler
+
+        super().__init__(opt)
+
+        # Input args
+        os.makedirs(self.output_folder, exist_ok=True)
+        self.worker_block_list = opt['worker_block_list'].split(',')
+
+        # Save paths
+        self.worker_results_path = os.path.join(
+            self.output_folder, 'worker_results.csv'
+        )
+        self.unacceptable_worker_ids_path = os.path.join(
+            self.output_folder, 'unacceptable_worker_ids.txt'
+        )
+        self.win_rate_by_date_path = os.path.join(
+            self.output_folder, 'win_rates_by_date.csv'
+        )
+        self.stat_mean_length_by_date_path = os.path.join(
+            self.output_folder, 'stat_mean_length_by_date.csv'
+        )
+        # TODO: do we really need to keep saving the mean length of these stats by date?
```

Comment:
Ha, I'd just remove this TODO, because there's no real reason to remove this feature now that we have it